src/debugAdapter/goDebug.ts

```diff
@@ -995,49 +995,67 @@ class GoDebugSession extends DebugSession {
 	protected variablesRequest(response: DebugProtocol.VariablesResponse, args: DebugProtocol.VariablesArguments): void {
 		verbose('VariablesRequest');
 		let vari = this._variableHandles.get(args.variablesReference);
-		let variables;
+		let variablesPromise: Promise<DebugProtocol.Variable[]>;
+		let loadChildren = async (exp: string, v: DebugVariable) => {
+			if (v.len > 0 && v.children.length === 0) {
+				// from https://github.com/derekparker/delve/blob/master/Documentation/api/ClientHowto.md#loading-more-of-a-variable
+				await this.evaluateRequestImpl({ 'expression': exp }).then(result => {
+					const variable = this.delve.isApiV1 ? <DebugVariable>result : (<EvalOut>result).Variable;
+					v.children = variable.children;
+				});
+			}
+		};
 		if (vari.kind === GoReflectKind.Array || vari.kind === GoReflectKind.Slice) {
-			variables = vari.children.map((v, i) => {
-				let { result, variablesReference } = this.convertDebugVariableToProtocolVariable(v, i);
-				return {
-					name: '[' + i + ']',
-					value: result,
-					evaluateName: vari.fullyQualifiedName + '[' + i + ']',
-					variablesReference
-				};
-			});
+			variablesPromise = Promise.all(vari.children.map(async (v, i) => {
```

Comment:
nit: does it makes sense to extract the code for each of these cases in separate smaller functions ?

Reply:
We could but since the functions are only relevant to `variablesRequest()` I think it make sense to keep them contained within it (now perhaps refactoring variable requests into its own class would clean this up...).